crates/ide/src/navigation_target.rs

```diff
@@ -153,6 +153,22 @@ impl NavigationTarget {
         )
     }
 
+    pub(crate) fn from_expr(
```

Comment:
Hm actually I think this is the wrong way to go about this, let's construct these manually in the relevant modules and instead expose the `orig_range_with_focus` functions to the rest of the crate

Reply:
So, should I implement `keywords_to_nav` in the relevant module, rather than directly calling `orig_range_with_focus_r` in `from_expr`?

Reply:
No let's put that into the calling module as well, not here I'd say

Reply:
If we move `from_expr` to the mod `goto_definition`, we will have to make `from_syntax` and `orig_range_with_focus_r` be `pub(crate)`, which seems a bit strange. 👀